index.js

```diff
@@ -1 +1,123 @@
+    regionLi.innerHTML = `<strong>Region: </strong>${country.region}`;
+    const capitalLi = document.createElement("li");
+    capitalLi.innerHTML = `<strong>Capital: </strong>${country.capital}`;
+
+    ulElement.appendChild(populationLi);
+    ulElement.appendChild(regionLi);
+    ulElement.appendChild(capitalLi);
+
+    countryInfoDiv.appendChild(titleElement);
+    countryInfoDiv.appendChild(ulElement);
+
+    countryLink.appendChild(countryFlagDiv);
+    countryLink.appendChild(countryInfoDiv);
+
+    countryLink.addEventListener("click", () => {
+        window.location.href = countryLink.href;
+    });
+
+    return countryLink;
+}
+
+let countries = [];
+
+const requestCountries = async () => {
+    try {
+        const response = await fetch("https://restcountries.com/v3.1/all");
+        if (!response.ok) {
+            throw new Error('Network response was not ok.');
+        }
+        const data = await response.json();
+        countries = data;
+    
+        const countriesGrid = document.querySelector(".countries-grid");
+    
+        countries.forEach(country => {
+            const countryBlock = createCountry(country);
+            countriesGrid.appendChild(countryBlock);
+        });
+    } catch (error) {
+        console.error('Error fetching country data:', error);
+    }
+ }
+
+requestCountries();
+
+const dropdownWrapper = document.querySelector('.dropdown-wrapper');
+const dropdownHeader = document.querySelector('.dropdown-header');
+
+dropdownHeader.addEventListener('click', () => {
+    dropdownWrapper.classList.toggle('open');
+});
+
+const dropDownItems = document.querySelectorAll('.dropdown-body li');
+dropDownItems.forEach(item => {
+    item.addEventListener('click', () => {
+        const selectedRegion = item.getAttribute('data-region');
+        filterCountriesByRegion(selectedRegion);
+        dropdownWrapper.classList.remove('open');
+    });
+});
+
+const updateDisplay = (countries) => {
+    const countriesGrid = document.querySelector(".countries-grid");
+    countriesGrid.innerHTML = "";
+
+    countries.forEach(country => {
+        const countryBlock = createCountry(country);
+        countriesGrid.appendChild(countryBlock)
+    });
+}
+
+const filterCountriesByRegion = (region) => {
+    if (region === "all") {
+        updateDisplay(countries);
+    } else {
+        const filteredCountries = countries.filter(country => country.region === region);
+        updateDisplay(filteredCountries);
+    }
+}
+
+const searchInput = document.querySelector('.search-input');
+
+searchInput.addEventListener('input', function() {
+    const filteredValue = this.value.toUpperCase();
+    const countryLinks = document.querySelectorAll('.country');
+
+    countryLinks.forEach(countryLink => {
+        const countryName = countryLink.getAttribute('data-country-name').toUpperCase();
+        if(countryName.includes(filteredValue)){
+            countryLink.style.display = '';
```

Comment:
No display?

Reply:
leave it empty,  so it displays them, unlike countryLink.style.display = 'none'; which came right after :)